src/main/java/teammates/logic/api/Logic.java

```diff
@@ -1587,15 +1592,13 @@ public String getFeedbackSessionResultSummaryInSectionAsCsv(String courseId,
      * Preconditions: <br>
      * * All parameters are non-null.
      */
-    public boolean hasStudentSubmittedFeedback(String courseId, String feedbackSessionName, String studentEmail)
+    public boolean hasStudentSubmittedFeedback(FeedbackSessionAttributes fsa, String studentEmail)
             throws InvalidParametersException, EntityDoesNotExistException {
```

Comment:
Wait, I'm not exactly talking about the right stuff. The one I talked about above is the method that follows, i.e `isFeedbackSessionCompletedByStudent`. Still, this Logic method is changed as an (successful) attempt to reduce redundant queries.